Felipe - felipegermanos/Week 10/fgc-week_10.do

```diff
@@ -0,0 +1,191 @@
+
+/* First, we drop all rows that don't contain department level data and
+ clean the department names */
+
+keep if regex(NOMCIRCONSCRIPTION, "DEPARTEMENT")
+
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, "DEPARTEMENT", "",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " DE ", "",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " DU ", "",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " DES ", "",.)
+replace NOMCIRCONSCRIPTION =  subinstr(NOMCIRCONSCRIPTION,"D' ","",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " D'", "",.)
+
+
+replace NOMCIRCONSCRIPTION = lower(strtrim(strtrim(NOMCIRCONSCRIPTION)))
+gen department = NOMCIRCONSCRIPTION
+drop NOMCIRCONSCRIPTION
+
+/* Looking at the data, it is possible to find one typo in the department names */
+replace department="arrha" if department=="arrah"
+
+
+/* Finally, we save the temporary file*/
+tempfile density
+save `density'
+
+
+/* Here is the code we use to merge the datasets on "department" */
+use "$civ_section0", clear 
+decode b06_departemen, generate(department)
+merge m:1 department using `density'
+
+
+/* Finally, we drop empty rows (case when merge ==2) */
+drop if _merge==2 
+drop _merge
+
+
+/* And this exports the dataset */
+save question_1, replace
+
+
+
+
+
+********************************************************************************
+******************************   Question 2   **********************************
+********************************************************************************
+
+/* We also covered most of this algorythm in class - ordering latitude and longitude
+and choosing appropriate clusters following such ordering. Notice that we need to 
+use a package, geodist, to run this code, so there might be an error because of it */
+
+
+use "$gps_data", clear
+
+
+/* We start by generating a scatterplot to check for clusters of observations */
+sort latitude longitude
+scatter longitude latitude, title("Q2 - Clusters by latitude and longitude")
+
+/* Next, we divide the data, which is sorted, in blocks of six, to reflect
+the fact that each field worker is assigned to 6 households, and use geodist to 
+check how efficient our ordering was. Notice nothing in the code is specific to 
+this dataset, so it should be generalizible */
+egen field_worker = seq(), block(6) 
+bysort field_worker (longitude latitude): gen lat_shifted = latitude[_n+1]
+bysort field_worker (longitude latitude): gen longit_shifted = longitude[_n+1] 
+geodist latitude longitude lat_shifted longit_shifted, generate(distance)
+
+/* Finally, we can try to measure how well we did by summing the total distance
+travelled */
+
+bysort field_worker: egen distance_travelled = total(distance)  
+sum distance_travelled 
+
+
+tabulate distance_travelled // all enumerators are walking under 2 miles
+summarize distance_travelled // enumerators are walking under 1 mile on average
+

```

Comment:
There is a really useful program "sepscatter" that can produce a scatter plot displaying points differently based on a third variable. I used it to run "sepscatter latitude longitude, separate(field_worker)" to create a plot depicting the coordinates of each household, but with different shapes/colors based on the enumerator assigned to the house. This will give you a good visual of how far enumerators are traveling and how logical your assignment was!

Just a neat program I'd consider taking a quick look at.